src/lib/y2storage/btrfs_subvolume.rb

```diff
@@ -109,6 +126,67 @@ def can_be_auto_deleted=(value)
       save_userdata(:can_be_auto_deleted, value)
     end
 
+    # Size of the referenced space of the subvolume, if known
+    #
+    # The information is obtained from the qgroup of level 0 associated to the subvolume,
+    # which implies it can only be known if quotas are enabled for the filesystem.
+    # If quota support is disabled, this method returns nil.
+    #
+    # If the filesystem already existed during probing but got no quota support enabled at
+    # that moment, this method returns zero if quota support was enabled after probing.
+    #
+    # @return [DiskSize, nil] nil if quota support is disabled, zero if it was enabled only
+    #   after probing, the known size in any other case
+    def referenced
```

Comment:
AFAIS, when there is no quota or no qgroup:

~~~
#referenced #=> nil
#exclusive #=> nil
#referenced_limit #=> unlimited
#exclusive_limit #=> unlimited
~~~

I would expect to either return a size or nil in all these methods. Maybe it does not make sense to use those values when there is no quota, so maybe these methods should return nil in that case. What do you think?

Reply:
If I'm querying the `#referenced_limit`, I want to know what the max size is. And I want a consistent response to that in all situations (ie. `unlimited`if there is no limit). I didn't ask about the reason. **So far**, there are two situations in which the size is unlimited: quotas are disabled or quotas are enabled but a limit was not set. When asking about the limit, I don't care in which situation I am. I only want to know what the limit is (`unlimited` if none).

If I want to know the status of quotas there are other methods for that.

The case of `referenced` is different. It returns `nil` simply because there is no information. Not to indicate that quotas are disabled.